code/go/0chain.net/chaincore/chain/protocol_view_change_integration_tests.go

```diff
@@ -46,11 +54,16 @@ func (c *Chain) SetupSC(ctx context.Context) {
 				case reg := <-isRegisteredC:
 					if reg {
 						logging.Logger.Debug("SetupSC - node is already registered")
+						func() {
+							defer func() {
+								recover()
+							}()
+							close(doneC)
```

Comment:
it's a bad logic to catch error from close(), you need to close channels only on writer side and close it once